iconSVG add icon/current family feature

bwmilby opened the pull request:
- convert several space indents to tab

Fixed/added docs:
- iconListMatching() - return LCB list of icon names that match a search string
- iconData() - return LCB array of default family icon data

Added the following handlers:
- iconFamilies() - returns a list of current icon families in the library
- iconNamesForFamily() - returns a string list of icon names for a specified family
- iconListForFamily() - returns a LCB list of icon names for a specified family
- iconDataForFamily() - returns LCB array of icon data for a specified family
- addIcon() - add an icon to the iconsvg library
- addIconsForFamily() - add a family of icons to the iconsvg library
- getCurrentIconFamily() - get the family used as default (i.e. for iconPicker)
- setCurrentIconFamily() - set the family to be used as default (i.e. for iconPicker)
- deleteIconFamily() - remove a family from the library
- iconArrayMatchingInAllFamilies() - search for icon matches among all families

bwmilby committed:
iconSVG improvements

- convert several space indents to tab

Fixed/added docs:
-iconListMatching() - return LCB list of icon names that match a search string
-iconData() - return LCB array of default family icon data

Added the following handlers:
-iconFamilies() - returns a list of current icon families in the library
-iconNamesForFamily() - returns a string list of icon names for a specified family
-iconListForFamily() - returns a LCB list of icon names for a specified family
-iconDataForFamily() - returns LCB array of icon data for a specified family
-addIcon() - add an icon to the iconsvg library
-addIconsForFamily() - add a family of icons to the iconsvg library

bwmilby committed:
iconSVG library - current family

Add the ability to get/set the current icon family in the iconSVG library.  The initial value is "fontawesome" (previous behavior).  The "ide" family is loaded when the library is initialized and is available immediately.  If the user adds additional families, they will also be available.  This will allow the existing `iconpicker` to be used to view icons from any family that is loaded.

bwmilby committed:
IconSVG library - fix iconListMatching documentation

Missing parameter in documentation of iconListMatching added

livecodeali commented:
I wonder if `iconListMatching()` should keep its existing behavior and we should add something like `iconArrayMatchingInAllFamilies()` that returns an array keyed by family with a list of matching icons for each one. If there are lots of families, then the returned list from `iconListMatching()` might be somewhat frustrating as then you have to iterate over the list to figure out what family they are in

bwmilby commented:
@livecodeali I'll do an update tonight.  I'm thinking we need `private _iconListMatching(pSearch, pFamily)` internally.  We could then make `iconListMatching()` just return for the current family as suggested.  The new function would use the private function to generate a list for each family for the array.

livecodeali commented:
Yep, sounds good to me!

bwmilby committed:
iconSVG (update matching code, delete family added)

- general documentation cleanup for the new updates and addition of families in general
- added a note to `iconSVGPathFromName` and `iconCodepointFromName` describing how to include family with name
- add `deleteIconFamily` handler
- add `iconArrayMatchingInAllFamilies` handler
- reverted `iconListMatching` handler to just search current family
- added private handler `_iconListMatching` which does the search on a specific family

BerndN commented on extensions/libraries/iconsvg/iconsvg.lcb:
```diff
@@ -1617,16 +1624,21 @@ public handler iconSVGPathFromName(in pName as String) returns String
-Related: iconNames(handler)
+>Note:  The icon name is accepted in the form "name" or "family/name".  
+If no family is specified, then the current family is searched first.  
+If not found, other families are serched for the provided name.
```
should be searched

bwmilby replied:
thanks

bwmilby committed:
iconSVG library docs

spelling correction (serched -> searched)

livecodeali commented:
@bwmilby Looks great!
@livecode-vulcan review ok de7b785

livecode-vulcan commented:
💙 review by @livecodeali ok de7b785628f59a1fb0c4190f9be4a94e14045d6b

livecode-vulcan commented:
😎 test success de7b785628f59a1fb0c4190f9be4a94e14045d6b

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/1144)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/1160)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/1134)
* try-community-universal-ios-iphoneos11.2: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos11.2/builds/87)
* try-community-universal-ios-iphonesimulator11.2: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator11.2/builds/87)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/1139)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/1161)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/1144)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/1160)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/1143)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/1153)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/1177)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1138)
